server/datastore/mysql/scripts.go

```diff
@@ -239,3 +239,77 @@ WHERE
 	}
 	return scripts, metaData, nil
 }
+
+func (ds *Datastore) GetHostScriptDetails(ctx context.Context, hostID uint, globalOrTeamID uint, opt fleet.ListOptions) ([]*fleet.HostScriptDetail, *fleet.PaginationMetadata, error) {
+	type row struct {
+		ScriptID    uint       `db:"script_id"`
+		Name        string     `db:"name"`
+		HSRID       *uint      `db:"hsr_id"`
+		ExecutionID *string    `db:"execution_id"`
+		ExecutedAt  *time.Time `db:"executed_at"`
+		ExitCode    *int64     `db:"exit_code"`
+	}
+
+	sql := `
+SELECT
+	s.id AS script_id,
+	s.name,
+	hsr.id AS hsr_id,
+	hsr.created_at AS executed_at,
+	hsr.execution_id,
+	hsr.exit_code
+FROM
+	scripts s
+	LEFT JOIN (
+		SELECT
+			id,
+			host_id,		
+			script_id,
+			execution_id,
+			created_at,
+			exit_code
+		FROM
+			host_script_results r
+		WHERE
+			host_id = ?
+			AND NOT EXISTS (
+				SELECT
+					1
+				FROM
+					host_script_results
+				WHERE
+					host_id = ?
+					AND id != r.id
+					AND script_id = r.script_id
+					AND(created_at > r.created_at
+						OR(created_at = r.created_at
+							AND id > r.id)))) hsr
```

Comment:
That's the sub-select to make sure we only return the latest execution of that particular script for the host, right? Nothing to worry about at the moment, but if we start seeing performance issues with many scripts, maybe we could add a `latest_execution` flag column and set it when we record a new result (and unset any previously existing one for the same host+script ids) to speed up reads.